### DISCUSSION TITLE
docker

### DISCUSSION THREAD

**Discussion by I-am-all** [bol-van/zapret#236] [Ideas]
я писал сам разные программы под линукс, но  (не хочу обидеть) такую запутанную для установки программу редко встречал.  (думаю сегодня часов 5 просидел пробовал на убунту поставить: откуда конфиги берутся непонятно, зачем ставить какие то обновлялки непонятно., где включается режим работы и их описание - непонятно)
может проще докер образ для программы сделать? 

ну и (сам имел отношения к документации и знаю что не каждому дано) так сложно и путанно все описывать , при этом перегружать какими то отстраненными размышлениями технический текст - это у автора талант, конечно. Вам бы поручить писать инструкции для тех, кто настраивает блокировки, чтобы они не справились  и бросили :) 
ко мне иногда люди по работе подходят помочь написать текст. я долго читаю и пишу одним простым предложением то, что они писали на страницу. они удивляются и говорят что талант (у меня всегда 3 по русскому и сочинению были)
вы правда считаете что docs/quick_start.txt это быстрый старт? :) это больше похоже на рассказ про программу за бутылкой пива. 

если често  тупо ВПР с прокси поставить и все. обфусицировать его чем нибудь, чтобы не узнали и работать так.

**Comment by bol-van** [comment]
напишите проще
причем так, чтобы потом не доставали с вопросами типа кау скопировать файл
или я сделал и у меня не заработало

цель специально убрать людей, которые не готовы вникать и хотят простую таблетку
ее просто не существует. это мысль номер один
придется знать, разбираться, пробовать
не готов ? мимо

**Reply by I-am-all** [reply]
если разберусь как поставить и заставить работать. пока не потянул. вероятно, глуп. Буду копить силы. 
я пытался разбираться - поэтому и писал про документацию. так и не нашел понятного объяснения чем отличают разные режиме (filter  от tpws; tpws полностью прозрачный или нет, применяются ли в нем параметры и какие, где подгружается конфиг, где тупо ip адрес прописать на котором слушается порт,  и многие другие)

Я без претензий, но три популярный решения byeDPI, goodbyDPI и yotubeUnblock за день собрал, поставил, оттестировать все с разными параметрами, посмотрел все Wireshark-ом,  для части создал докер образы для быстрого переноса. 

спасибо за труд!!!

**Reply by bol-van** [reply]
у гудбая тоже все не гладко
народ перебирает простые шаблоны
и потом половина кричит у меня не работает
еще часть не знает как работать в консоли и куда вбивать более подробные параметры

я создавал не решение для чайников, а гибкий инструмент для борьбы с дпи
можете выкинуть все кроме tpws и nfqws
разберитесь в механизмах обхода
подбирайте
знайте как работать с иптаблес

**Reply by bol-van** [reply]
и поймите главное. это борьба. это хак. это не комерческий проект, где программе никто не сопротивляется и она удобно работает
это мои рассуждения без пива правда
но это суть что условия везде разные
факторов много
все меняется
рецепта раз два нет

**Reply by I-am-all** [reply]
> иптаблес

не знаю как по имени обратиться, отличный пример. универсальный инструмент работы с пакетами - лопатить ядро. Знаю как непосредственный участник разработки первых российских DPI. :-) тогда мы думали, что это надо для тарификации трафика.  пробовали программно на уровне ядра, иногда чуть выше ядра. Потом стали думать аппаратные. Потом хуавей с циско пришел. потом РДП ру этот появился со своими сборками. 
я это к тому, что если для обхода надо лезть так глубоко то это уже саморазработанное решение.  проще вникнуть в суть и самому с нуля писать. 
я так сделал когда мне не нравились приложения для интервальных тренировок для смартфона (задает режимы под конкретный вид тренировки - спарринг, силовая, HIIT). Изучил разработку под Андроид, попросил помощи у ИИ и уже долгое время почти каждый день пользуюсь своим приложением. 
короче, не хочется становиться разработчиком вариантов обхода :-)
лучше другим чем то заняться. 
Хотелось  скачать бинарник, поставить его в systemctl, открыть порты в iptables и смотреть видео.

**Reply by I-am-all** [reply]
> я создавал

не обижайтесь. я так , вероятно, переключаю обиду на себя (что оказался там где мне блокируют что я хочу смотреть) на окружающий мир. Спасибо за старались сделать!!!  это прекрасно. что есть люди которые создают что-то.  
да местный РТ имеет какое то особенное средство блокировки.

**Reply by I-am-all** [reply]
а можно, пользуясь случаем, узнать, что такое "посылы" ( --split-pos=<offset>	; делить все **посылы** на сегменты в указанной позиции.)?
 вроде используются  и "пакеты" и "запросы".  Непонятно тогда, что такое "посылы"

и еще вопрос, если можно. tpws и nfqws отличаются чем то, кроме того, как в них поступает трафик.? tpws это прокси, а nfqws берет пакеты из очереди, я так понял

**Reply by bol-van** [reply]
посылы это не пакеты. это вызовы процесса   send или write  в сокет
дальще ос их разбивает на пакеты

тпвс работает на уровне сокетов tcp
с пакетами не работает
но на практике если включить tcp_nodelay, ос выплевывает все сразу
потому если посыл отослать в 2 вызова  send, мы получим  тцп сегментацию
split pos  делит посыл, что обычно сводится к разделению 1 сегмента на 2
либо разделению в середине одного из сегментов, если посыл более 1 пакета

nfqws  берет уже сформированные ос пакеты и издевается над ними
меняет, дропает, шлет свои

**Comment by bimawa** [comment]
Да ну, за такую докаументацию - мое почтение! <3

**Comment by NOFEXtreme** [comment]
Ожидал увидеть обсуждения по Docker, но, без обид, кто-то здесь считает себя непризнанным гением: проблема не в его непонимании, а в «плохой» инструкции.

@samvova, если вы действительно такой опытный специалист, вам не составило бы труда взять исходники nfqws или tpws, собрать их и настроить свои ipt. Всё остальное, это удобная оболочка для работы с этими инструментами. Легко критиковать, но сделать что-то полезное для других требует не только знаний, но и времени и терпения. Не устраивает инструкция? Напишите более понятную! Если уж так превозносите свои умения.

@bol-van, спасибо тебе за труд!

**Comment by bol-van** [comment]
Я уже миллион раз озвучивал свою позицию.
Этот инструмент работает лишь в приложении к понимающей голове.
Я понимаю : многие - врачи, педагоги и прекрасные няшки, и им хочется усталым после работы нажать на кнопку, чтобы посмотреть ютубе. Но ситуация в принципе не предполагает простого решения. Даже если оно появится, и им начнут массово пользоваться, реакция системы не заставит себя долго ждать. Поэтому сложность - скорее плюс, чем минус.
Проблема лежит не в технической плоскости, а скорее в вопросе почему у нас мир такой. Почему вы, врачи, педагоги и прекрасные няшки, вовремя ничего не сделали, чтобы мы жили в раю.
Цель доки - не сделать максимально возможно легкую кнопку, а дать возможность изучить вопрос как это работает. Остальное, как говорится, приложится. Иначе будет как обезьяна с очками, что и демонстрирует реальный опыт эксплуатации.
Сделать простое решение на базе обмана DPI может быть и можно, но это во-первых потребует экспоненциального роста усилий со стороны разработчиков, во-вторых неизбежно привлечет сопротивление системы и еще больше потребует усилий на его купирование. В итоге система все равно свое возьмет, только это еще больше ее подстегнет к ужесточению.
А усилия будут спущены в унитаз.
